distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1543,6 +1543,196 @@ public void testFederatedDownloadProductToCacheOnlyCacheDisabled() throws Except
+            Integer numActivities) throws Exception {
+        expect("Waiting for notifications").within(10, SECONDS)
+                .until(() -> cometDClient.getMessages(NOTIFICATIONS_CHANNEL)
+                        .size() == numNotifications);
+        expect("Waiting for activities").within(10, SECONDS)
+                .until(() -> cometDClient.getMessages(ACTIVITIES_CHANNEL)
+                        .size() == numActivities);
+
+    }
+
+    private void validateNotificationsSuccess(CometDClient cometDClient, String filename,
+            String resourceData) throws Exception {
+
+        List<String> notifications = cometDClient.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        assertThat(notifications.size(), is(1));
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(0)),
+                filename,
+                getResourceRetrievalCompletedMessage(resourceData.length()),
+                "complete");
+
+    }
+
+    private void validateActivitiesSuccess(CometDClient cometDClient, String filename,
+            String resourceData) throws Exception {
+        List<String> notifications = cometDClient.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        assertThat(notifications.size(), is(1));
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(0)),
+                filename,
+                getResourceRetrievalCompletedMessage(resourceData.length()),
+                "complete");
+
+    }
+
+    private void validateNotificationsFailure(CometDClient cometDClient, String filename)
+            throws Exception {
+        List<String> notifications = cometDClient.getMessagesInAscOrder(NOTIFICATIONS_CHANNEL);
+        assertThat(notifications.size(), is(4));
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(0)),
+                filename,
+                "Resource retrieval retrying after 1 bytes. Attempt 1 of 3.",
+                "retrying");
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(1)),
+                filename,
+                "Resource retrieval retrying after 1 bytes. Attempt 2 of 3.",
+                "retrying");
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(2)),
+                filename,
+                "Resource retrieval retrying after 1 bytes. Attempt 3 of 3.",
+                "retrying");
+        CometDMessageValidator.verifyNotification(JsonPath.from(notifications.get(3)),
+                filename,
+                "Resource retrieval failed. Unable to retrieve product file.",
+                "failed");
+
+    }
+
+    private void validateActivitiesFailure(CometDClient cometDClient, String filename)
+            throws Exception {
+
+        List<String> activities = cometDClient.getMessagesInAscOrder(ACTIVITIES_CHANNEL);
+        assertThat(activities.size(), is(8));
```

Comment:
See above comment. Technically we should have a private method generating these batch responses based on a provided constant, but in this case the likelihood of changing the test circumstances / expectation parameters is slim.